sql/ts/src/skdb_util.ts

```diff
@@ -102,3 +104,42 @@ export class SKDBTable extends Array<Record<string, any>> {
     });
   }
 }
+
+/* ***************************************************************************/
+/* Transaction API for programmatically constructing and executing
+   transactions against an SKDB instance */
+/* ***************************************************************************/
+
+export class SKDBTransaction {
+  private query_params: Params;
+  private stmts: Array<string>;
+  private db_handle: SKDB;
+
+  constructor(skdb: SKDB) {
+    this.query_params = {};
+    this.stmts = [];
+    this.db_handle = skdb;
+  }
+
+  addStatement(stmt: string): SKDBTransaction {
+    this.stmts.push(stmt.trim().replace(/;$/, ""));
+    return this;
+  }
+
+  params(params: Params): SKDBTransaction {
```

Comment:
It might be a lot of work I guess and I worry about how robust it would be, but it would be really cool if addStatement took the sql and a params object. Then for each query you could specify params independently. The logic here could then do variable renaming to ensure there's no conflict. Maybe there's a global params call too for shared params. Not sure how valuable this is but thought I'd throw out the idea.

Reply:
I thought about this (and discussed a bit with Lucas+Josh) but decided against having a `Params` argument to `addStatement` to avoid variable renaming/conflict woes, as this is all happening before any sql parsing.

My intuition is that transactions will often reuse the same param across multiple queries/statements, where a single shared param namespace is desirable.  And I worry about the conceptual complexity of separate per-query params and transaction-wide params with different semantics.

But, one can also imagine wanting to write something like

```
for(const user_id in users)
   txn.addStatement(
     "update users set ... where id=@user_id",
     {user_id}
   )
```

which is not possible in this interface. The parsing and renaming of variables to support that wouldn't be too bad -- just scan for @-variables, taking care to avoid substitutions within string literals.

Reply:
Agree on all fronts.

> taking care to avoid substitutions within string literals.

This - and I'm sure there are other situations - is what I was worried about. Probably end up writing a parser.